Description: Ignore source errors when using the default doclet with Java 9
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: no
--- a/src/main/org/apache/tools/ant/taskdefs/Javadoc.java
+++ b/src/main/org/apache/tools/ant/taskdefs/Javadoc.java
@@ -1858,6 +1858,12 @@
 
         doTags(toExecute);
         doSource(toExecute);
+
+        if (doclet == null && LanguageLevel.isDebianBuild() && JavaEnvUtils.isAtLeastJavaVersion("9")) {
+            toExecute.createArgument().setValue("--ignore-source-errors");
+            log("Debian build on Java 9+ detected: Adding the --ignore-source-errors option");
+        }
+
         doLinkSource(toExecute);
         doNoqualifier(toExecute);
 		
